open-feature/dart-server-sdk
# feat(eventing): add support for state change reactions

## This PR
- Adds eventing feature to react to state changes in the provider or flag system.



### Notes
This feature includes event hooks for providers and flag changes.

### Follow-up Tasks
- Add tests for edge cases in the eventing system.

### How to test
- Run the project.
- Simulate provider or flag changes to confirm the eventing feature triggers correctly.
